lua/typescript-tools/api.lua

```diff
@@ -27,4 +27,32 @@ function M.request_diagnostics(callback)
   }, callback)
 end
 
+--- Returns an |lsp-handler| that filters TypeScript diagnostics with the given codes.
+--- <pre>lua
+--- local api = require('typescript-tools.api')
+--- require('typescript-tools').setup {
+---   handlers = {
+---     -- Ignore 'This may be converted to an async function' diagnostics.
+---     ['textDocument/publishDiagnostics'] = api.filter_diagnostics { 80006 }
+---   }
+--- }
+--- </pre>
+---
+---@param codes integer[]
+function M.filter_diagnostics(codes)
+  return function(err, res, ctx, config)
+    local filtered = {}
+    local with_lookups = vim.deepcopy(codes)
```

Comment:
I think we don't need to copy, this list will be always(I think?) constant list of codes

Reply:
We need the copy because the reverse lookup mutates the input table, and so when invoking the handler repeatedly there will be an error because the reverse key mapping will already be there from a previous invocation.

Reply:
Hmm, but if you move this line to upper scope - line between 42 and 43 `with_lookups` will be initialized once per run of configuration and for all other calls of inner function "edited" table will be in cloasue. I think it should work like that, but it is untested idea :D

Reply:
Hmmm that’s true. I’ll try it out :)